aicsimageio/writers/ome_zarr_writer.py

```diff
@@ -301,12 +330,14 @@ def write_image(
                     if zdimindex > -1
                     else 1
                 )
-                chunk_dim_map["Z"] = nplanes_per_chunk
-                chunk_dim_map["Y"] = lasty
-                chunk_dim_map["X"] = lastx
-                chunk_dims.append(
+                chunk_dim_map[DimensionNames.SpatialZ] = nplanes_per_chunk
```

Comment:
i wonder if we could check the passed in chunk_dims and do something smarter here?  I have a couple possible ideas.  
1. compute a chunk dims based on the user-provided one that is proportional to each dimension by the amount of the downsampling
2. compute a chunk dims that is proportional but has the same memory size per chunk (is this desirable?)
3. let users pass in an array of chunk_dims tuples, one for each downsampled level if they want to. (can users pass in pre-downsampled data?)

Reply:
I just committed a basic version of bullet 1 here [906e823](https://github.com/AllenCellModeling/aicsimageio/pull/519/commits/906e82388d6e20345e06d7b95dd99ce468334db4). With each level, we reassess X and Y around the scaler.downscale and change that to be the new chunk_dims.